arlas-core/src/main/java/io/arlas/server/core/FluidSearch.java

```diff
@@ -754,6 +758,31 @@ private ValuesSourceAggregationBuilder setAggeragatedGeometryStrategy(Aggregatio
         return aggregationBuilder;
     }
 
+    private ValuesSourceAggregationBuilder setHitsToFetch(Aggregation aggregationModel, ValuesSourceAggregationBuilder aggregationBuilder) throws ArlasException {
+        if (aggregationModel.fetchHits != null) {
+            TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(FETCH_HITS_AGG);
+            Integer size = Optional.ofNullable(aggregationModel.fetchHits.size).orElse(1);
+            topHitsAggregationBuilder.size(size);
+            List<String> includes = new ArrayList<>();
+            if (aggregationModel.fetchHits.include != null) {
+                for (String field : aggregationModel.fetchHits.include) {
```

Comment:
You muse a .map instead of a list that you feed from a for() loop

Reply:
In addition to feeding the list, there is also an action of sorting on a field which can't be in a map